dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java

```diff
@@ -255,4 +240,29 @@ public static void addRepositoriesToRequest(CollectRequest collectRequest) {
       collectRequest.addRepository(repository);
     }
   }
+
+  /**
+   * Returns Maven repository specified as {@code mavenRepositoryUrl}, after validating the syntax
+   * of the URL.
+   *
+   * @throws IllegalArgumentException if the URL is malformed for a Maven repository
```

Comment:
URISyntaxException might be more appropriate here, or perhaps pass a URI/URL instead.

Reply:
This `URI` instance is just for validating the input string and the `URI` class (or `URL`)  is not used by Maven's `RemoteRepository`. So I'd say IllegalArgumentException is fine.